### ISSUE TITLE
What's the usage of ec_public::version_?

### ISSUE THREAD

**Issue by GingerMoon** [libbitcoin/libbitcoin#946]
Hi @evoskuil ,
There seems to be a bug related to ec_public == 
    bool b1 = (privateKey == privateKey1); // true
    bool b2 = (publicKey == publicKey1); // false -> only the "version_" fields are different.

After checking the source code, it's a little strange that the "version_" field only appears in three member methods:
operator=, operator==, version().
So I doubt this field is initialized randomly, which means that the operator== will always return false except the scenario of operator=.

**Comment by evoskuil** [comment]
Yes, the version_ field is dead code. Both it and the version() method should be removed. Please issue a PR. Thanks!
